src/targets/cocoapods.ts

```diff
@@ -24,7 +24,7 @@ const DEFAULT_COCOAPODS_BIN = 'pod';
 const COCOAPODS_BIN = process.env.COCOAPODS_BIN || DEFAULT_COCOAPODS_BIN;
 
 /** Options for "cocoapods" target */
-export interface CocoapodsTargetOptions extends TargetConfig {
+export interface CocoapodsTargetOptions {
```

Comment:
What's the main motivation behind this?

Reply:
No motivation, just pleasing TypeScript. Once I made the `name` field required, all these configs started to give type errors, complaining about a missing `name` attribute. Upon inspection, I realized none of these were actually extending `TargetConfig`, there was nothing shared. These were internal configs. What needed to be the `TargetConfig` is the one we pass in the constructor, which I refactored already.